float64[3] position
float64[3] rotation
float64[6] velocities